.github/workflows/upgrade-python-requirements.yml

```diff
@@ -0,0 +1,68 @@
+name: Upgrade Requirements
+
+on:
+  schedule:
+    # will start the job at 00:15 UTC every Tuesday
+    - cron: "15 0 * * 2"
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: "Target branch to create requirements PR against"
+        required: true
+        default: 'master'
+
+jobs:
+  upgrade_requirements:
+    runs-on: ubuntu-20.04
+
+    strategy:
+      matrix:
+        python-version: ["3.8"]
+
+    steps:
+      - name: setup target branch
+        run: echo "target_branch=$(if ['${{ github.event.inputs.branch }}' = '']; then echo 'master'; else echo '${{ github.event.inputs.branch }}'; fi)" >> $GITHUB_ENV
```

Comment:
Let me preface this by saying I am just now learning about github actions: Question, why would you specify inputs.branch.default = 'master' but then set 'master' again here manually?

Reply:
this adds the ability to trigger this workflow manually and specify a different branch than `master` to run the workflow
as you can see here: https://github.com/edx/edx-platform/actions/workflows/upgrade-python-requirements.yml
if the workflow is run by corn schedule it'd run the workflow against `master`